src/Halogen/Menu/Component.purs

```diff
@@ -0,0 +1,124 @@
+module Halogen.Menu.Component where
```

Comment:
Actually, re: module naming, in the notebook stuff at least I'd been following a scheme like this:

```
Thing.Component (which re-exports the other two)
Thing.Component.State
Thing.Component.Query
```

As the `State` and `Query` stuff is specific to the component, whereas you might have other non-component-related modules you want inside `Thing` (like data structures used in the state but are not the state/model itself).

I'm not too sure how great it is yet though, as it did get a little confusing over time, but I think that will happen any time we have a lot of modules with similar names.